### PR TITLE
Solution_for_engineers

### PR DESCRIPTION
**Pull request by AndriiFedoryshyn**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,54 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    skills = []
+
```
Comment (OleksandrPozden) [inline app/main.py:6]: skills should be inside constructor

File: app/main.py [hunk 2]
```diff
@@ -1 +1,54 @@
...
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+
+    skills = ["JavaScript", "HTML", "CSS"]
```
Comment (OleksandrPozden) [inline app/main.py:15]:
```suggestion
class FrontendDeveloper(SoftwareEngineer):
    def __init__(self, name: str) -> None:
        super().__init__(name)
        self.skills += ["JavaScript", "HTML", "CSS"]
```
And the same for others

**Review by OleksandrPozden — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (AndriiFedoryshyn) [reply app/main.py:15]: Sorry, but I can't do that, because in this case I didn't passed the tests. Please, check README

File: app/main.py (hunk 1 above)
Comment (Serhii-Leonenko) [reply app/main.py:6]: Fix it. Skills should be the instance's attribute

File: app/main.py (hunk 2 above)
Comment (Serhii-Leonenko) [reply app/main.py:15]: You can do this if you fix skills attribute according to previous comment

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (Serhii-Leonenko) [reply app/main.py]: It isn't fixed

File: app/main.py [hunk 3]
```diff
@@ -22,8 +20,7 @@ def create_awesome_web_page(self) -> str:
 class BackendDeveloper(SoftwareEngineer):
     def __init__(self, name: str) -> None:
         super().__init__(name)
-
-    skills = ["Python", "SQL", "Django"]
+        self.skills = ["Python", "SQL", "Django"]
```
Comment (Serhii-Leonenko) [inline app/main.py:23]: you should append skills with `+=` as was said to you in previous comments.

File: app/main.py [hunk 4]
```diff
@@ -44,9 +40,8 @@ def create_smooth_mobile_app(self) -> str:
 class FullStackDeveloper(FrontendDeveloper, BackendDeveloper):
     def __init__(self, name: str) -> None:
         super().__init__(name)
-
-    skills = ["Python", "SQL", "Django",
-              "JavaScript", "HTML", "CSS", ]
+        self.skills = ["Python", "SQL", "Django",
+                       "JavaScript", "HTML", "CSS"]
```
Comment (Serhii-Leonenko) [inline app/main.py:44]: you can remove the __init__ in this class if you fix skills append according to previous comments

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]

**Comment by AndriiFedoryshyn** [comment]
Дякую за перевірку.

File: app/main.py [hunk 5]
```diff
@@ -1 +1,47 @@
...
+        self.skills += ["Java", "Android studio"]
+
+    def create_smooth_mobile_app(self) -> str:
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(FrontendDeveloper, BackendDeveloper):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
```
Comment (ArsenPidhoretskyi) [inline app/main.py:42]: Redundant code

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**